pkg/promtail/client/client.go

```diff
@@ -238,6 +239,12 @@ func (c *client) send(ctx context.Context, buf []byte) (int, error) {
 	req = req.WithContext(ctx)
 	req.Header.Set("Content-Type", contentType)
 
+	// If the orgID is not empty promtail is running in multi-tenant mode, so
+	// we should send it to Loki
+	if orgID != "" {
+		req.Header.Set("X-Scope-OrgID", orgID)
+	}
+
 	resp, err := c.client.Do(req)
```

Comment:
Good point. Is it required to introduce authentication at this stage? Adding multi-tenant support may already cover the case of a private Loki installation with multi-tenant enable and without authentication.

Reply:
I'm wondering if this is something we plan to support. If someone wants to send to multiple Grafana Cloud account using a single Promtail deployment.

We can open an issue for following up later.

@slim-bean @gouthamve @jtlisi  What do you think ? You've got more experience on that matter with Cortex, does it seems useful ?

Reply:
We wouldn't be able to trust someone's provided `X-Scope-OrgID` header, without doing some authentication/authorization against them anyway (which is what we do to select the correct header now)  So there isn't really much value for this if you have an untrusted segment in your communication link.

I think the use cases where this does add value are referenced in the above linked issues, where someone controls all pieces of the infrastructure and want to store separate logs as separate tenants in Loki (maybe different groups within the same organization)

Reply:
My concern here is so far the client send basic auth when configured. Now we support multiple tenant but still one auth.

I'm definitely fine to not take this into account for that PR. But I'm just wondering how would that work if someone use this with Grafana Cloud for instance ? How can he provide token for each of his tenant he is outputting.

Reply:
_Given I would suggest to keep per-tenant auth for a future work (once we'll have at least someone asking for it), let me share some thoughts about it._

The auth is binded to a client instance. We currently have `MultiClient` which allows to push all logs to 1+ Loki clusters: a `MultiClient` internally can have multiple `Client`, 1 for each Loki cluster.

The proposed changes in this PR simply set the `X-Scope-OrgID` header differently based on the tenant, within the same instance of `Client`. This wouldn't work with a per-tenant authentication, because a per-tenant authentication would require different `Client` instances.

An idea would be introducing a third level of clients: `MultiClusterClient` (what is `MultiClient` today) -> `MultiTenantClient` (new - always push to the same client, but internally lazily instantiates a new `Client` for each tenant) -> `Client` (single tenant client).